wasm-thumbnail-js/src/lib.rs

```diff
@@ -0,0 +1,57 @@
+use std::error::Error;
+
+use wasm_bindgen::prelude::*;
+
+use image;
+use image::imageops::FilterType;
+use image::DynamicImage;
+use image::GenericImage;
+use image::GenericImageView;
+use image::ImageOutputFormat;
+
+/// Resize the input image specified by pointer and length to nwidth by nheight,
+/// returns a pointer to nsize bytes that containing a u32 length followed
+/// by the thumbnail bytes and padding
+#[wasm_bindgen]
+pub fn resize_and_pad(
+    src: &[u8],
+    nwidth: u32,
+    nheight: u32,
+    nsize: usize,
+) -> Result<Box<[u8]>, JsValue> {
+    let mut out: Vec<u8> = Vec::with_capacity(nsize);
+    // Reserve space at the start for length header
+    out.extend_from_slice(&[0, 0, 0, 0]);
+
+    if let Ok(thumbnail_len) = _resize_and_pad(src, &mut out, nwidth, nheight, nsize) {
+        out.splice(..4, thumbnail_len.to_be_bytes().iter().cloned());
+        return Ok(out.into_boxed_slice());
+    } else {
+        return Err(JsValue::from_str("bad"));
```

Comment:
not good - ideally we should convert the `Box<dyn Error>` to a `JsValue` but I wasn't sure how best to do that.